language-plutus-core/src/Language/PlutusCore/Name.hs

```diff
@@ -85,39 +84,40 @@ newtype Unique = Unique { unUnique :: Int }
 -- | The unique of a type-level name.
 newtype TypeUnique = TypeUnique
     { unTypeUnique :: Unique
-    }
+    } deriving (Eq)
 
 -- | The unique of a term-level name.
 newtype TermUnique = TermUnique
     { unTermUnique :: Unique
-    }
-
--- | The default implementation of 'HasUnique' for newtypes.
-newtypeUnique
-    :: (Wrapped new, HasUnique (Unwrapped new) unique', Coercible unique' unique)
-    => Lens' new unique
-newtypeUnique = _Wrapped' . unique . coerced
+    } deriving (Eq)
 
 -- | Types which have a 'Unique' attached to them, mostly names.
-class Coercible Unique unique => HasUnique a unique | a -> unique where
+class Coercible unique Unique => HasUnique a unique | a -> unique where
     unique :: Lens' a unique
+    -- | The default implementation of 'HasUnique' for newtypes.
+    default unique
```

Comment:
I think these days we can write this as an instance for `Newtype` and a `deriving via` clause on the types that need the instance. I think that's better than the weird special default implementations.

Reply:
I think it's fine to have a default implementation for a common use case when there is no other default implementation for another common use case in sight. I mean, it's simpler than making a newtype and then requiring the user to write `DerivingVia`. I'm not quite opposed to the latter and we can make a nice meta `ANewtype` `newtype` and use it here and maybe in other places and maybe extract it in a separate library and put it on hackage and make an announcement and become popular and rich, but I think my version is simpler, is equally sensible and doesn't sound like a plan from the Pinky and the Brain show.

I'm fine either way, so your call.

Reply:
Oh yes, `Newtype` is a class, not a type. Yes, in this instance we would need a `NewtypeUnique` type with an instance.

I guess my objection is that the `default` implementation allows you to have an implementation for *one* case where you implement something. But e.g. if you wanted to have an implementation for things that are newtypes and things that are `Generic`, then you can't do both. Whereas you can have multiple types for `deriving via` just fine.

Reply:
Yeah, my objection targets precisely your objection :)

I'm not too wild on having a default implementation via `Generic`, but a default implementation via a newtype does seem like a sensible default and I do not think that the potential `Generic` default is a good competitor.